### PR TITLE
#259 : Add comment with Redirect URL and Redirect URL pattern next to Client Secret field in Stores Configuration

### PR DESCRIPTION
**Pull request by konarshankar07**
### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#259: Add comment with Redirect URL and Redirect URL pattern next to Client Secret field in Stores Configuration

### CODE REVIEW

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@konarshankar07 thanks for the pull request. However, the idea is to generate redirect_uri and pattern for admin user based on callback controller url

**Comment by konarshankar07** [comment]
@sivaschenko 

Sorry, I think I haven't understood this task properly. So want to save the redirect url path on system config field?

**Comment by sivaschenko** [comment]
@kronarshankar the expected values shown to admin user should look like:

redirect url: `https://<magento_host>/<admin_path>/adobe_stock/oauth/callback`
redirect url pattern: `https://<magento_host>/<admin_path>/adobe_stock/oauth/callback/`

Where `<magento_host>` and `<admin_path>` are taken from the database, also the url have to include index.php if "Use webserver rewrites" is turned off and secret key if it's turned off.

Simply saying the `UrlInterface` have to be used to get the url that will be shown to admin user

File: AdobeStockAsset/Model/System/Config/Comment.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    /**
+     * @param string $elementValue
+     * @return string
+     */
+    public function getCommentText($elementValue)
+    {
+        $redirect_url = $this->url->getUrl('adobe_stock/oauth/callback');
+        $redirect_url_pattern = $this->url->getUrl('admin/adobe_stock/oauth/callback/');
```
Comment (konarshankar07) [inline AdobeStockAsset/Model/System/Config/Comment.php]: Even though I have added `admin` in the redirect_url_pattern but it showing
image

Comment (sivaschenko) [reply AdobeStockAsset/Model/System/Config/Comment.php]: Please use camelcase for variables names, i.e. `$redirectUrl`.

Redirect url pattern requires escaping, I think you can simply take `$redirectUrl` and add `\\` before each dot, i.e. `$redirectUrlPattern = str_replace('.', '\\.', $redirectUrl)`.

File: AdobeStockAsset/Model/System/Config/Comment.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        $commentText = sprintf("Configure an Adobe Stock account on the <a href=\"https://console.adobe.io/\" target=\"_blank\">Adobe.io</a> site to retrieve an Private key (Client secret).</br>
+        In order to create an integration in console.adobe.com Magento admin needs to enter %s and %s.", $redirect_url, $redirect_url_pattern);
```
Comment (sivaschenko) [inline AdobeStockAsset/Model/System/Config/Comment.php]:
```suggestion
        Note: the Default redirect URI is %s and the redirect URI pattern is %s.", $redirectUrl, $redirectUrlPattern);
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@konarshankar07 well done! just a couple of comments

File: AdobeStockAsset/Model/System/Config/Comment.php (hunk 1 above)
Comment (sivaschenko) [reply AdobeStockAsset/Model/System/Config/Comment.php]: sorry, there is no need to add `admin` to the redirect pattern, that was my typo

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**